File: BackEnd/apiFunctions/billOfMaterial.php
<?php
//*************************************************************************************************
// FileName : billOfMaterial.php
// FilePath : apiFunctions/
// Author   : Christian Marty
// Date		: 21.11.2023
// License  : MIT
// Website  : www.christian-marty.ch
//*************************************************************************************************
declare(strict_types=1);
global $database;
global $api;

if($api->isGet(Permission::BillOfMaterial_List))
{
    $query = <<< QUERY
        SELECT 
            BillOfMaterialNumber,
            Name,
            Description
        FROM billOfMaterial
    QUERY;
    $result = $database->query($query);
    \Error\checkErrorAndExit($result);

    foreach($result as &$item) {
        $item->ItemCode = \Numbering\format(\Numbering\Category::BillOfMaterial, $item->BillOfMaterialNumber);
        $item->Description = $item->Description??'';
    }
    $api->returnData($result);
}

File: API/billOfMaterial.py
# *************************************************************************************************
# FileName : billOfMaterial.py
# Author   : Christian Marty
# Date     : 16.02.2024
# License  : MIT
# Website  : www.christian-marty.ch
# *************************************************************************************************

class BillOfMaterial(object):
    def __init__(self, eerp_instance):
        self._eerp = eerp_instance

    def list(self):
        data = self._eerp.get('billOfMaterial')
        return data
